Dockerfile

```diff
@@ -2,18 +2,23 @@ FROM ruby:3.0.3-alpine
 RUN apk add gcc git libc6-compat libc-dev make nodejs postgresql13-dev sqlite-dev tzdata yarn
 
 WORKDIR /myapp
-COPY Gemfile /myapp/Gemfile
-COPY Gemfile.lock /myapp/Gemfile.lock
+
+COPY Gemfile /myapp/
+COPY Gemfile.lock /myapp/
 RUN gem update --system
 RUN bundle install
-RUN gem install nokogiri --platform=ruby
-COPY package.json /myapp/package.json
-COPY yarn.lock /myapp/yarn.lock
-RUN yarn
+
+COPY package.json /myapp/
+COPY yarn.lock /myapp/
+RUN yarn install --frozen-lockfile
+
 ADD . /myapp
+ENV RAILS_ENV=production \
+    RAILS_SERVE_STATIC_FILES=yes \
+    LANG=en_GB.UTF-8 \
+    SECRET_KEY_BASE=TestKey
```

Comment:
Here we're generating the docker image using this temporary key, but when we deploy later, are we using a different `SECRET_KEY_BASE` in the application environment?

---

[More background on what SECRET_KEY_BASE does](https://stackoverflow.com/questions/25426940/what-is-the-use-of-secret-key-base-in-rails-4/25427302#25427302).

TL;DR it can be anything, including a randomly generated GUID that we don't have to know / retrieve ever. It shouldn't live in an encrypted storage like KeyVault because we should only write it once and then forget about it.

When it gets replaced, all session cookies become invalid, so it shouldn't be changed between routine deployment (it should be persistent).

It should also be different between all envs.

Reply:
used terraform random_uuid to pass in env variable on deployment